internal/impl/aws/input_kinesis.go

```diff
@@ -545,6 +549,33 @@ func isShardFinished(s *kinesis.Shard) bool {
 	return *s.SequenceNumberRange.EndingSequenceNumber != "null"
 }
 
+// Lists all the shards in the stream and iterate through the shards as the API has a limit of 1000 shards per call
+func (k *kinesisReader) getShardsResult(streamID string) (*kinesis.ListShardsOutput, error) {
+	listShardsInput := kinesis.ListShardsInput{
+		StreamName: aws.String(streamID),
+	}
+	if k.listShardsAtLatest {
+		listShardsInput.SetShardFilter(&kinesis.ShardFilter{Type: aws.String("AT_LATEST")})
+	}
+
+	totalShardRes, err := k.svc.ListShardsWithContext(k.ctx, &listShardsInput)
+	if err != nil {
+		return totalShardRes, err
+	}
+	nextToken := totalShardRes.NextToken
+	for nextToken != nil && *nextToken != "" {
```

Comment:
nit: It's recommended to use aws type conversion helper methods when using aws sdks in golang (https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/aws#ToString)

```suggestion
	for aws.ToString(nextToken) != "" {
```

Reply:
Thanks! Updated to `StringValue` as we are using aws v1